content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx

```diff
@@ -0,0 +1,530 @@
+
+```ts
+const order = await prisma.order.update({
+  where: {
+    id: 10
+  }
+  data: {
+    shippingAddress: {
+      // Update just the zip field
+      update: {
+        zip: "41232"
+      }
+    }
+  }
+})
+```
+
+For an optional embedded type, like the `billingAddress`, use `upsert` to create a new record if it does not exist, and update the record if it does:
+
+```ts
+const order = await prisma.order.update({
+  where: {
+    id: 10
+  }
+  data: {
+    billingAddress: {
+      // Create the address if it doesn't exist,
+      // otherwise update it
+      upsert: {
+        set: {
+          street: "1084 Candycane Lane",
+          city: "Silverlake",
+          zip: "84323"
+        },
+        update: {
+          zip: "84323"
+        }
+      }
+    }
+  }
+})
+```
+
+You can also use the `unset` method to remove an optional embedded type. The following example uses `unset` to remove the `billingAddress` from an `Order`:
+
+```ts
+const order = await prisma.order.update({
+  where: {
+    id: 10
+  }
+  data: {
+    billingAddress: {
+			// Unset the billing address
+      // Removes "billingAddress" field from order
+      unset: true
+    }
+  }
+})
+```
+
+### Changing multiple composite types
+
+Use the new `set`, `push` `updateMany` and `deleteMany` methods to change a list of composite types:
+
+- `set`: Set an embedded list of composite types, overriding any existing list
+- `push`: Push values to the end of an embedded list of composite types
+- `updateMany`: Update many composite types at once
+- `deleteMany`: Delete many composite types at once
+
+For example, use `push` to add a new photo to the `photos` list:
+
+```ts
+const product = prisma.product.update({
+  where: {
+    id: 10,
+  },
+  data: {
+    photos: {
+      push: [{ height: 100, width: 200, url: '1.jpg' }],
+    },
+  },
+})
+```
+
+Use `updateMany` to update photos with a `url` of `1.jpg` or `2.png`:
+
+```ts
+const product = prisma.product.update({
+  where: {
+    id: 10,
+  },
+  data: {
+    photos: {
+      updateMany: {
+        where: {
+          url: '1.jpg',
+        },
+        data: {
+          url: '2.png',
+        },
+      },
+    },
+  },
+})
+```
+
+The following example uses `deleteMany` to delete all photos with a `height` of 100:
+
+```ts
+const product = prisma.product.update({
+  where: {
+    id: 10,
+  },
+  data: {
+    photos: {
+      deleteMany: {
+        where: {
+          height: 100,
+        },
+      },
+    },
+  },
+})
+```
+
+## Deleting composite types with <inlinecode>delete</inlinecode> and <inlinecode>deleteMany</inlinecode>
+
+To remove records which embed a composite type, use the `delete` or `deleteMany` methods. This will also remove the embedded composite type.
+
+For example, use `deleteMany` to delete all products with a `size` of `"Small"`. This will also delete any embedded `photos`.
+
+```ts
+const deleteProduct = await prisma.product.deleteMany({
+  where: {
+    size: 'Small',
```

Comment:
Is this correct? In the schema it's `sizes  Size[]`

Reply:
Oops, yep. Does that look correct now?

Reply:
Yep, looks good now!